core/domain/skill_domain.py

```diff
@@ -1776,3 +1776,72 @@ def from_dict(cls, skill_mastery_dict):
             skill_mastery_dict['skill_id'],
             skill_mastery_dict['degree_of_mastery']
         )
+
+
+class CategorizedSkills:
+    """Domain object for representing categorized skills' ids and
+    descriptions.
+    """
+
+    def __init__(self):
+        """Constructs a CategorizedSkills domain object."""
+        self.categorized_skills = {}
+
+    def add_topic(self, topic_name):
+        """Adds a topic to the categorized skills and initializes its value
+        as an empty dict.
+
+        Args:
+            topic_name: str. The name of the topic.
+        """
+        self.categorized_skills[topic_name] = {}
```

Comment:
Program defensively; raise error if the topic name already exists.

Why not initialize uncategorized skills at this point?

Reply:
Done. Initializing all subtopic skills and uncategorized skills at one place now